service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -194,6 +194,16 @@
     private final JudgeAndLegalAdvisor judgeAndLegalAdvisor;
     private final C2DocumentBundle temporaryC2Document;
     private final List<Element<C2DocumentBundle>> c2DocumentBundle;
+
+    @JsonIgnore
+    public C2DocumentBundle getLastC2DocumentBundle() {
+        return Optional.ofNullable(c2DocumentBundle)
```

Comment:
Don't think you need to wrap it in an optional, if you want to do the chaining logic then the first line can be `Stream.of(ElementUtils.unwrapElements(c2DocumentBundle)).filter....`